File: 209/rust/src/main.rs
fn main() {}

pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
    let (mut left, mut right, mut sum, n) = (0, 0, 0, nums.len());
    let mut ans = n + 1;
    while right < n {
        while right < n && sum < target {
            sum += nums[right];
            right += 1;
        }
        while sum >= target {
            ans = usize::min(ans, right - left);
            sum -= nums[left];
            left += 1;
        }
    }
    if ans == n + 1 {
        ans = 0;
    }

    ans as i32
}

#[cfg(test)]
mod tests {
    use crate::min_sub_array_len;

    #[test]
    pub fn test1() {
        let ansr = min_sub_array_len(7, [2, 3, 1, 2, 4, 3].to_vec());
        assert_eq!(ansr, 2)
    }
}

File: 209/python/main.py
def minSubArrayLen(target, nums):
    left = right = curr = 0
    ans = float('inf')

    while right < len(nums):
        curr += nums[right]

        while curr >= target:
            curr -= nums[left]
            ans = min(ans , right - left + 1)
            left += 1
        right += 1
    return ans if ans < float('inf') else 0



if __name__ == '__main__':
    print(minSubArrayLen(7, [2,3,1,2,4,3]))




















# Example 1:

# Input: target = 7, nums = [2,3,1,2,4,3]
# Output: 2


# Example 2:

# Input: target = 4, nums = [1,4,4]
# Output: 1

# Example 3:

# Input: target = 11, nums = [1,1,1,1,1,1,1,1]
# Output: 0